### ISSUE TITLE
choosing who owns & maintains curriculum resources

### ISSUE THREAD

**Issue by jsms90** [foundersandcoders/master-reference#271]
# CALLING ALL ALUMNI WHOSE WORKSHOPS ARE PART OF THE FAC CURRICULUM

Thank you so much to everyone who contributed to #205!! :heart_eyes: 

I'm really sorry for suggesting that you all had to follow the one option that I gave you! Multiple, perfectly valid options have emerged, depending on:
+ who you consider the owner to be
+ whether the author wants to continue maintaining the workshop themselves, or hand over the responsibility

[What does "being a maintainer" entail?](https://github.com/foundersandcoders/master-reference/issues/271#issuecomment-304247597#what-does-being-a-maintainer-entail)

## Tasks & their deadlines
The following tasks need to be completed before FAC10 & FACN1 start their curriculum planning weeks.

### Authors
_Regardless of whether you have already transferred ownership of your repo_, please choose one of the following options.  
(You can choose a different option for each repo you have made for the `master-reference`)

| Option | Owner | Maintainer |
|-----------|:---------------|:---------------|
| 1 | `foundersandcoders` | TBC (someone else in the FAC community) |
| 2 | `foundersandcoders` | original author                                               |
| 3 |   original author         | original author                                                |

+ Add the option you have chosen to [the table below](https://github.com/foundersandcoders/master-reference/issues/271#issuecomment-304245255), for _each_ one of your repos - **by Monday 29th May**
+ Follow [the guidance below](https://github.com/foundersandcoders/master-reference/issues/271#issuecomment-304246990) on how to enact these choice - **by Tuesday 30th May**

### Me
**By Tuesday 30th May**
Use the answers provided in the table below, to:
+ [x] record these options in a document within our `curriculum-planning` folder

**By Wednesday 31st May** (still waiting on some answers)
+ [x] update all the schedule files in `coursebook`, to link to the repos that have been mirrored / transferred / forked
+ [x] add `AUTHOR.md` files to all the repos under `foundersandcoders`

EDIT: Removed the following point, given #280
+ <s>add much more detail to the [planning of weeks doc](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/planning-weeks.md) & link to the appropriate sections of our `CONTRIBUTING.md` guide</s>

**Comment by jsms90** [comment]
Please edit this comment & add either `1`, `2` or `3` to the "Option" column

### Week 1
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [Accessibility workshop](https://github.com/foundersandcoders/web-accessibility) | @jsms90 |    |
| [Accessibility challenge](https://github.com/njsfield/accessibility-challenge) | @njsfield |     |
| [Learn git basics](https://github.com/foundersandcoders/learn-git-basics) | @jsms90 |    |
| [CSS gallery challenge](https://github.com/njsfield/css-gallery-challenge) | @njsfield |    |

### Week 2
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [Learn QUnit](https://github.com/skibinska/learn-qunit) |@skibinska |   |
| [Fizz buzz](https://github.com/skibinska/fizzbuzz) | @skibinska |    |
|[Romanizer](https://github.com/skibinska/romanizer) | @skibinska |    |
| [Waterfall function workshop](https://github.com/RhodesPeter/waterfall-function-workshop) | @RhodesPeter |    |
| [Pure functions easy testing](https://github.com/Jwhiles/pure-functions-easy-testing) | @Jwhiles |    |

### Week 3
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [API workshop](https://github.com/lucymonie/api-workshop) | @lucymonie |    |
| [XHR workshop](https://github.com/skibinska/xhr-workshop) | @skibinska |    |
| [Workshop APIs](https://github.com/emilyb7/workshop-APIs) | @emilyb7 |    |
| [Workshop client-side design](https://github.com/eliascodes/workshop-client-side-design) | @eliascodes |    | 
| [Parallel function morning challenge](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-3/morning-challenge.md) (move into separate repo) | @emilyb7 ? |    |

### Week 4
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [Node intro workshop](https://github.com/bradreeder/Node-Intro-Workshop) | @bradreeder |    |
| [ES6 morning challenge](https://github.com/stevehopkinson/es6-challenge) | @stevehopkinson |    |
| [tape testing](https://github.com/matthewglover/tape-testing) | @matthewglover |    |
| [modularising a simple node server](https://github.com/shouston3/modules-challenge) | @shouston3 |    |

### Week 5
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [server with shot and tape](https://github.com/njsfield/tdd-node-server-with-shot-and-tape) | @njsfield |    |
| [error handling workshop](https://github.com/njsfield/error-handling-workshop) | @njsfield |    |
| [node shell workshop](https://github.com/msachi/Node-Shell-Workshop) | @msachi (forked version - original is in `foundersandcoders`, author has requested that we point students to particular exercises from the original workshop rather than creating a new workshop that is a combination of the existing workshops) |   |
| [request module workshop](https://github.com/RhodesPeter/request-module-workshop) | @RhodesPeter |    |

### Week 6
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [postgres workshop](https://github.com/FAC9/postgres-workshop) | @tbtommyb |    |
| [pg walkthrough](https://github.com/shiryz/pg-walkthrough) | @shiryz |    |
| [pg workshop](https://github.com/shiryz/pg-workshop) | @shiryz |    |
| [db morning challenge](https://github.com/shiryz/db-morning-challenge) | @shiryz |    |

### Week 7
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [hapi intro](https://github.com/msachi/hapi-intro) | @msachi |    |
| [handlebars intro workshop](https://github.com/marisid/handlebars-intro-workshop) | @marisid |    |
| [handlebars morning challenge](https://github.com/Jbarget/handlebars-morning-challenge) | @Jbarget |    |
| [injector](https://github.com/SavageWilliam/injector) | @SavageWilliam (fork - sufficient divergence from the original => original author has requested that this be turned into its own workshop) |    |
| [hapi auth cookie workshop](https://github.com/foundersandcoders/hapi-auth-cookie-ws) | @SavageWilliam |    |

### Week 8
| Resource | Author | Option |
|-----------|:---------------|:---------------|
| [OAuth workshop](https://github.com/foundersandcoders/oauth-workshop) | @jsms90 |   |
| [JWT workshop](https://github.com/foundersandcoders/jwt_workshop) | @denesnori |    |

**Comment by emilyb7** [comment]
@jsms90 done. The original parallel challenge is based on the one we did with @besarthoxhaj during FAC9. I definitely can't claim to be the author. For FAC10 I wrote a different version based on the GitHub API (don't think it was ever merged into the master reference). I've added a link to that above.

**Comment by jsms90** [comment]
### Guidance

**Comment by jsms90** [comment]
### FAQ

Q: What does "being a maintainer" entail?

A: Please follows these open source guides on [Best Practices for Maintainers](https://opensource.guide/best-practices/) & [Building a Community](https://opensource.guide/building-community/).  

Maintainers should expect spikes in contribution during the following weeks:
+ the week that their repo is part
+ curriculum planning week

Remember: these are run during 3 different weeks - one for each of the current campuses.

**Comment by RhodesPeter** [comment]
I mirror @emilyb7, I didn't create the waterfall function workshop. It was @des-des who created it, my version was a simplified version based on FAC9s reaction to the [original](https://github.com/des-des/morning-challenges).

**Comment by jsms90** [comment]
@emilyb7 So what's [this](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-3/morning-challenge.md)? :confused: 

@RhodesPeter Could you still indicate the option you would prefer, if your version is the one that continues to be used?
Is there sufficient divergence from Eoin's original material, that this can be considered your own work? Or have you spoken to him about whether he would prefer to receive a PR to his original?

**Comment by emilyb7** [comment]
@jsms90 it's something I copied from Bes. Original challenge was on repl.it and I can't find the link right now... I don't consider myself to be the author. I think if the next group of mentors decide to continue running this challenge, they can take charge of making a repo for it.

**Comment by jsms90** [comment]
@emilyb7 Sure thing. So what did FAC10 use? Bes' one that you copied into `master-reference`, or your one that's on your own repo? Because the one on your profile isn't linked to from the `master-reference` schedule for week 3 :confused:

**Comment by emilyb7** [comment]
The GitHub one...
> For FAC10 I wrote a different version based on the GitHub API (don't think it was ever merged into the master reference)
It was a very last minute thing. There should be a reference in an issue somewhere. Nazareth used Bes's version.

**Comment by jsms90** [comment]
@SavageWilliam Considering you have chosen option 3 for your [injector workshop](https://github.com/SavageWilliam/injector), could you please follow the request of the original author. Instead of forking his repo, could you please copy the current content of your repo into a new one, that is not forked from his?

**Comment by jsms90** [comment]
@SavageWilliam sorry for the change. Just dug back into my gitter history and I got it wrong. @JMurphyWeb said to make a PR to his original.

![image](https://cloud.githubusercontent.com/assets/15656538/26562446/375bb972-44be-11e7-9413-4b15ccf6d9ce.png)

@JMurphyWeb is that still what you would prefer?

**Comment by des-des** [comment]
I think maybe whoever owns the repository is the owner, if its not a fork.

That person should give credit in the readme to the original creator

**Comment by jsms90** [comment]
@des-des :+1: I'm going to be creating `AUTHOR.md` files for all the repos under `foundersandcoders`. It's in my little todo list at the top. It's something that @alexis-l8 and I talked about a couple of times and it seems there is a precedent for it in some open source projects.

As a side benefit, it could also make it easier to find out how to submit PRs to, rather than having to write a separate contributing guide for every single workshop / morning challenge.

**Comment by JMurphyWeb** [comment]
@jsms90 I am open to PRs on mine: https://github.com/JMurphyWeb/injector.. What needs to be done?

**Comment by jsms90** [comment]
Hey @JMurphyWeb 

The comment above is about a PR needing to be raised from @SavageWilliam's version to your own. Assuming that's what you want?

@SavageWilliam's version has substantial changes from your original. But his is the version in `master-reference`. The question is whether you want the updates that @SavageWilliam has made to your repo, to be merged into your own? Or whether you want his to exist as a separate, stand-alone project?

**Comment by jsms90** [comment]
@RhodesPeter You have marked the "request module workshop" as option 3, but it is still in `foundersandcoders`. Would you like to move it back to your own profile?

**Comment by JMurphyWeb** [comment]
Ok i'll take a better look into @SavageWilliam's.  Was there any feedback on Fac10's side? I heard FACN1's feedback on it.

**Comment by RhodesPeter** [comment]
@jsms90 yes please.

**Comment by jsms90** [comment]
@RhodesPeter You can do it :+1: 
I'm actually not sure whether I can (haven't tried)

Same process:
Settings --> Transfer --> type in your own profile instead of `foundersandcoders`

Students are reviewing the weeks as we speak, so please let me know when you're done and I'll update the `master-reference`, so that the link in the week's schedule isn't broken.

**Comment by RhodesPeter** [comment]
I can't find a way to do this. I will just change my number on this to 2.

**Comment by jsms90** [comment]
@RhodesPeter Ah! Sorry, I assumed you had transferred ownership, but you must have made a mirror, because there is already a "request module workshop" on your own profile. Is that correct? Is the `foundersandcoders` version a duplicate?

**Comment by RhodesPeter** [comment]
@jsms90 - I did transfer ownership. About 3 minutes ago I cloned my workshop from F&C, changed the remote and pushed up to my own GitHub.

**Comment by jsms90** [comment]
@RhodesPeter IOn principle,  think that the version that gets used by FAC needs to retain the full history. There are a couple of closed issues on the repo in FAC, one of which was raised by a student. The idea behind making a mirror, instead of just pushing up a clone, is to retain the history.

I have just transferred the repository that was sitting in `foundersandcoders` to my own profile - https://github.com/jsms90/request-module-workshop/ to make sure that this works.

If you want to take the workshop back into your profile, please let's keep the version that has the history :pray: Moving it back under your own profile is no problem. I'm sorry for the hassle, because I was the one that told you to transfer ownership in the first place, but we can do these.

Please do the following:
+ delete the cloned version you have just put in your own profile
+ In the version with the history (I will put it back in https://github.com/foundersandcoders/request-module-workshop/ now):
  + Go to "settings"
  + Go to the bottom and click "transfer"
  + Enter the name of the workshop i.e. `request-module-workshop`
  + Enter the name of your profile i.e. `RhodesPeter`

Should be fine :+1:

**Comment by jsms90** [comment]
@JMurphyWeb 
+ An issue called "Confusion with the 'Testing in hapi & server.inject workshop' (week 7 day 2)" #217 was raised when students were in week 7
+ There is mention of the workshop in [FACN1's stop go continue](https://github.com/foundersandcoders/london-curriculum/blob/master/stop-go-continue/fac-10/cohort-week-7.md)  
(No explicit mention of it in [London's sgc](https://github.com/foundersandcoders/nazareth-curriculum/blob/master/stop-go-continue/fac-n1/week7.md#stop))

FAC10 are planning the next curriculum as we speak. Unfortunately, I think we might be having this conversation a little too late :sweat: They need to know where to raise issues, and who to assign them to, and they will begin raising them either today or tomorrow.

But I appreciate that you haven't had much time to think about this, since you haven't had a PR to review until now, and Will has been the one responding to students', because he was the one in the classroom delivering it. It's also completely possible that students may decide to scrap this from the next curriculum altogether. So how about this for now:
+ we direct FAC10 & FACN1 to raise issues on Will's version and make PRs to the fork
+ once the changes are merged and the version going into the next curriculum is finalised, we let you know
+ you can have a look at the version as it stands then, and decide whether you would be interested in accepting a PR?
  + if you do accept a PR, you can then decide for yourself, which of the options in the flowchart you want for the repo
  + if you don't want a PR, Will can contain to be the maintainer (assumingyou are happy with this deviation from your material being used in a FAC curriculum?)

What do you think?

**Comment by RhodesPeter** [comment]
@jsms90 - I don't have access to the settings of the Founders & Coders GitHub account. You must have higher access than me.

**Comment by jsms90** [comment]
Gotcha. My bad :scream: Sorry! :sweat: 

I really don't know what I'm allowed to do in terms of making people owners of `foundersandcoders` - see https://github.com/foundersandcoders/london-membership/issues/38

But I'm sure it won't matter if I do it temporarily, so you can move your workshop back. Gonna do that now. Please let me know when you're done :+1:

**Comment by RhodesPeter** [comment]
![screen shot 2017-05-31 at 12 13 41](https://cloud.githubusercontent.com/assets/18164707/26629515/a0556f26-45fa-11e7-9683-66d5594b6a38.png)

**Comment by jsms90** [comment]
@RhodesPeter Haha damn. Ok. Made you an owner. Try now :grin:

**Comment by RhodesPeter** [comment]
@jsms90 done. You can change the permissions back now :)

**Comment by Jwhiles** [comment]
I'm alarmed by how complex this process has become. 

As I understand it the whole point of open source software is that anyone can fork from your work and change it in any way they see fit.

The obvious solution is for foundersandcoders to fork these repos and then for each cohort change them to suit their needs.

**Comment by NataliaLKB** [comment]
> As I understand it the whole point of open source software is that anyone can fork from your work and change it in any way they see fit.

From what I understand its more:

"As I understand it the whole point of open source software is that anyone can _contribute_, share and distribute, and fork and change."

But personally I find forking and changing without even attempting to contribute back just not respectful to the original work. But if the original author doesn't want those changes or is not responsive then totally okay to fork*.

But I agree completely this has become a complex "process" where I don't think it needs to be either. As long as everyone is respectful when dealing with open source things generally work out.


* @jsms90 pointed out to me in a different thread how you have to fork to make a change anyway, but I think the key difference is either you use your fork and don't communicate via issue or PR to the original author, or you do make a PR or at least open an issue about it  try and contribute back first).

**Comment by jsms90** [comment]
> The obvious solution is for foundersandcoders to fork these repos and then for each cohort change them to suit their needs.

This was my flawed thinking when I forked @NataliaLKB's repo in the first place. But I never would have done it, if I had found the time to create my own material. Still, it was a mistake, and the whole thing has been a deeply uncomfortable experience. I learned my lesson. Let us please _please_ learn from the mistakes of the past.

**If our students' needs diverge from the needs of everyone else who uses that external repo, students are better off creating their own new & different repo.** This is a last case resort. Better if we all find a way to make the best resource possible _for everyone_.

**Comment by jsms90** [comment]
> I'm alarmed by how complex this process has become.

My original request: for everything that has explicitly been made for the FAC curriculum to live under `foundersandcoders` is much simpler. To be honest, I'm extremely disappointed, that not all of our cohort considers this to be the entire point of the material's creation :cry:

**Comment by iteles** [comment]
The issue here seems to be that as cohorts slightly modify workshops we end up with scores of 'versions' of the same workshop - slightly different - across many people's repositories.

Unless one of those is the 'official' workshop, then they get lost in the ether and all of that great work is wasted because the Founders & Coders community doesn't get to see them - a real shame. 

I suggest we need:
+ A master list of the possible workshops that have been 'approved' for our curriculum (discussion around where these should live and consistency of curriculum across campuses are being dealt with in other issues)
+ If your workshop is part of the curriculum, you have to be willing to maintain and entertain the possibility of changes
+ If someone wants to simplify or complexify one of these workshops, they raise an issue on the original repo
  + If the changes are accepted, work should begin on these, _building on the original_ either by adding examples, clarifying the wording, adding more steps, splitting into beginner/intermediate/advanced, etc
  + If the changes are rejected by the creator (this shouldn't be an issue for repos under the Founders & Coders organisation), a simplified version can be created by that campus, under the organisation umbrella, _referring clearly to the original_ and noting that this was a simplified (for example) version (to be discouraged unless absolutely necessary)

This is all pretty intuitive to me and I completely agree with @NataliaLKB , this is open source, courtesy and common sense will get us far.

**Comment by Jbarget** [comment]
what @iteles ^^^ said

**Comment by jsms90** [comment]
@iteles Below is a list of how we are currently trying to implement all of your suggestions. Please let me know if you think these methods are adequate, or you can think of _anything_ we can do to improve or do differently :heart_eyes:

> A master list of the possible workshops that have been 'approved' for our curriculum

:+1: Our current list exists in an issue: #43. I think it is mine and @bradreeder's intention to go through this list properly at some point and turn it into an `md` file

> If your workshop is part of the curriculum, you have to be willing to maintain and entertain the possibility of changes

That's exactly what the flowchart [above](https://github.com/foundersandcoders/master-reference/issues/271#issuecomment-304246990) is supposed to help us to ascertain. It has been recorded in [this doc](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/creation-of-resources.md) already for permanency. I'm in the process of figuring out how to best communicate that to students. I think @njsfield has helped to reach [a better solution than I was originally suggesting](https://github.com/foundersandcoders/workshop-client-side-design/issues/5)

> If someone wants to simplify or complexify one of these workshops, they raise an issue on the original repo

I believe this is exactly what @bradreeder is telling FAC10 right now.

> (to be discouraged unless absolutely necessary)

Covered in https://github.com/foundersandcoders/master-reference/issues/298#issuecomment-305449937

**Comment by jsms90** [comment]
Submitted PRs to all "parts", which adds author and maintainer at the top. All are merged except 3, all of which relate to "parts" whose weeks are no longer in the curriculum. 

Closing the issue
